project-base/storefront/store/useCookiesStore.ts

```diff
@@ -1,27 +1,47 @@
 import { CookiesStoreContext } from 'components/providers/CookiesStoreProvider';
 import { useContext } from 'react';
+import { v4 as uuidV4 } from 'uuid';
 import { useStore } from 'zustand';
 import { createStore } from 'zustand/vanilla';
 
-export type CookiesStoreState = { lastVisitedProductsCatnums: string[] | undefined };
+export type CookiesStoreState = {
+    lastVisitedProductsCatnums: string[] | null;
+    userIdentifier: string;
+};
 
 export type CookiesStoreActions = {
     setCookiesStoreState: (value: Partial<CookiesStoreState>) => void;
 };
 
 export type CookiesStore = CookiesStoreState & CookiesStoreActions;
 
-export const defaultInitState: CookiesStoreState = {
-    lastVisitedProductsCatnums: undefined,
-};
+export const getDefaultInitState = (): CookiesStoreState => ({
```

Comment:
Yes. If its an object, all customers have the same user identifier. This would behave differently if we had used edge runtime, but for node, there is only one instance of the server.